includes/class-wc-payments-disputes-summary-cache.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * WC_Payments_Disputes_Summary_Cache class
+ *
+ * @package WooCommerce\Payments
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class which manages the payment dispute summary cache.
+ */
+class WC_Payments_Disputes_Summary_Cache {
```

Comment:
> I see we have a generic cache facility in [Database_Cache](https://github.com/Automattic/woocommerce-payments/blob/444e8fb10755415a43fb18b7ca2426b1ee3543be/includes/class-database-cache.php) - should we leverage that instead of managing direct?

I'll take a look at this.

Reply:
I've had a look at the [`Database_Cache`](https://github.com/Automattic/woocommerce-payments/blob/444e8fb10755415a43fb18b7ca2426b1ee3543be/includes/class-database-cache.php) class and it does work in this scenario. 

I've pushed [7b15862](https://github.com/Automattic/woocommerce-payments/pull/4326/commits/7b158629af74df457a8d39837e8603f2c1453f8c) which replaces the `WC_Payments_Disputes_Summary_Cache` uses with the `Database_Cache`. 

I'll add these notes to the PR description but I'll include them here too. 

- Using the cache means it automatically comes with transient like behaviour.
- By default the cache will last for [24 hours](https://github.com/automattic/woocommerce-payments/blob/fix/354/includes/class-database-cache.php#L303-L304) unless we specify otherwise. 
- `Database_Cache` supports a getter (`get_or_add()`) which will attempt to get the value from the cache, check the expiration, validate the value (via a `callable`). It will also set the value if it isn't valid or set. 
- The validation function isn't super useful in our situation but I have passed the `is_array` function incase the cache is corrupted.